### DISCUSSION TITLE
Remix.run useLoaderData returns type any

### DISCUSSION THREAD

**Discussion by otatar** [remix-run/remix#6470] [Q&A] [answered]
I was just trying this new thing remix.run and came across this. Everywhere in cos for this framework they are using const data = useLoaderData() to have type of data variable inferred. But for me this is not working:
````
import { json } from "@remix-run/node";
import type { LoaderFunction, LoaderArgs } from "@remix-run/node";
import { useLoaderData } from "@remix-run/react";
import { getUsers } from "~/models/users.server";

export const loader: LoaderFunction = async (args: LoaderArgs) => {
  const users = await getUsers();
  return json({ users: users });
};

export default function Users() {
  const { users } = useLoaderData<typeof loader>();
  return (
    <div>
      {users.map((u) => (
        <p key={u.id}>{u.username}</p>
      ))}
    </div>
  );
}
````
My VSCode says that variable users has any type and there is a squiggly line red line bellow u in map...what is wrong? package.json:
````
"@remix-run/css-bundle": "^1.16.0",
    "@remix-run/node": "^1.16.0",
    "@remix-run/react": "^1.16.0",
    "@remix-run/serve": "^1.16.0",
````

I asked this on stackoverflow, but with no asnwer:
https://stackoverflow.com/questions/76314884/remix-run-useloaderdata-returns-type-any

**Comment by machour** [comment]
@otatar The best thing when facing cases where it works for everybody but not for you, is to provide a small repository reproducing the issue.

In your loader, what does your editor say about `users` when it gets assigned `await getUsers()` ? Is it properly typed there?

**Reply by otatar** [reply]
![image](https://github.com/remix-run/remix/assets/22593698/07d9b0fb-f13d-44da-af3c-0aeb9ee54cb3)
It is typed correctly....just pulling things from prisma

**Reply by AlemTuzlak** [reply]
@otatar try removing the LoaderFunction and just let it infer the type

**Reply by AlemTuzlak** [reply, marked as answer]
so do:
```jsx
export const loader = async (args: LoaderArgs) => {
  const users = await getUsers();
  return json({ users: users });
};

```

**Reply by otatar** [reply]
Thank you "zemljak", that was it....I dont know from where I have got that LoaderFunction type :D

**Reply by AlemTuzlak** [reply]
no worries, happy to help! Small world @otatar. You can install Remix Forge in vs-code and generate up to date loaders via right click, that can help you speed up and avoid these kinds of hick-ups.

**Reply by otatar** [reply]
Thank you, definitely going to do that, and I will give a try to remix-hook-form...

**Reply by kiliman** [reply]
@otatar `LoaderFunction` was used in Remix prior to v1.4 (inferred types), so you may have seen that example somewhere.
